src/trace/implementations/mod.rs

```diff
@@ -80,15 +85,31 @@ where
     R: Semigroup+Clone,
 {
     type Key = K;
+    type KeyOwned = K;
     type Val = V;
     type Time = T;
     type Diff = R;
 }
 
+impl<K,V,T,R> Update for Box<((K, V), T, R)>
```

Comment:
This `Box<_>` thing is just a random type. I couldn't get `[_]` to work because the unsizedness infected everything. Also, `(([K], V), T, D)` was just very upsetting to Rust; it never got off the ground.